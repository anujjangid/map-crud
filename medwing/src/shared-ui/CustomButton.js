// @flow

import React from 'react';
import {
 TouchableOpacity, Text, StyleSheet, Dimensions 
} from 'react-native';

const { width } = Dimensions.get('window');

type Props = {
  text: string,
  onPress: () => void
};

const CustomButton = (props: Props) => {
  const { text, onPress } = props;
  return (
    <TouchableOpacity
      style={
        text === 'EDIT'
          ? styles.editTouchableOpacity
          : styles.deleteTouchableOpacity
      }
      onPress={onPress}>
      <Text style={styles.textStyle}>{text}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  textStyle: {
    fontSize: 20,
    color: '#ffffff',
    textAlign: 'center'
  },

  deleteTouchableOpacity: {
    position: 'absolute',
    bottom: 0,
    width,
    padding: 10,
    left: 0,
    zIndex: 999,
    alignItems: 'center',
    backgroundColor: '#F80505'
  },
  editTouchableOpacity: {
    position: 'absolute',
    bottom: 40,
    alignSelf: 'flex-end',
    width: width * 0.3,
    right: 0,
    padding: 10,
    zIndex: 100,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#33B2FF'
  }
});

export default CustomButton;
