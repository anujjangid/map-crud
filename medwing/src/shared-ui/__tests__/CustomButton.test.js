import React from 'react';
import renderer from 'react-test-renderer';
import CustomButton from '../CustomButton';

test('renders correctly', () => {
  const tree = renderer.create(<CustomButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
