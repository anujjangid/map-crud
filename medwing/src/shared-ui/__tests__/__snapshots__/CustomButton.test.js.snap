// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F80505",
      "bottom": 0,
      "left": 0,
      "opacity": 1,
      "padding": 10,
      "position": "absolute",
      "width": 750,
      "zIndex": 999,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  />
</View>
`;
